test/testdrive/webhook.td

```diff
@@ -186,6 +186,19 @@ z
 $ webhook-append name=webhook_validation_panic status=503
 abc
 
+# Dropping a webhook source should drop the underlying persist shards.
```

Comment:
I'd like to see a test that introduces some contention here for creating/using/dropping these sources--might be able to convince someone from @MaterializeInc/qa to help out.

Reply:
Ping some folks on the QA team!

Reply:
Chatted with QA, currently our parallel testing framework is `test/parallel-workload`, which concurrently runs a bunch of different SQL statements. But AFAICT what's missing from the framework is the ability to send POST requests to append to a webhook.

Instead I added a Rust test that spins up a few threads that continuously send data, then drop the source, wait a little bit to keep sending data, and assert we get an expected number of successes and not found responses. What do you think about this @sploiselle?